src/content/app/genome-browser/components/zmenu/Zmenu.tsx

```diff
@@ -51,15 +59,68 @@ const Zmenu = (props: ZmenuProps) => {
 
   const destroyZmenu = () => {
     dispatch(changeHighlightedTrackId(''));
-    setZmenus && setZmenus(pickBy(zmenus, (value, key) => key !== props.id));
+    setZmenus &&
+      setZmenus(pickBy(zmenus, (value, key) => key !== props.zmenuId));
   };
 
   const direction = chooseDirection(props);
   const toolboxPosition =
     direction === Direction.LEFT ? ToolboxPosition.LEFT : ToolboxPosition.RIGHT;
 
+  const { content, variety } = props.payload;
+
+  const transcripts = content.filter(
+    (f) => f.metadata.type === ZmenuFeatureType.TRANSCRIPT
+  ) as ZmenuContentTranscript[];
+  const genes = content.filter(
+    (f) => f.metadata.type === ZmenuFeatureType.GENE
+  ) as ZmenuContentGene[];
+
+  const features: (ZmenuContentTranscript | ZmenuContentGene)[] = [];
+
+  // get the first variety type
+  const firstVarietyType = variety[0].type;
```

Comment:
Didn't @ens-ds23 say that the contract is that only one variety is going to have a `type` field on it? If so, then we want to do `const zmenuType = variety.find(variety => Boolean(variety.type))?.type as string`

Reply:
That's done